tests/test_config.py

```diff
@@ -73,3 +90,162 @@ def test_reference_anchor_for_module(repo):
+    get_bmo_data.return_value = test_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    file_config.config["people"].append({"bmo_id": 3})
+    file_config.write()
+
+    assert len(file_config.config["people"]) == 5
+
+    # Check that the new entry is unaffected after write.
+    for person in file_config.config["people"]:
+        if person["bmo_id"] == 3:
+            # Ensure only the bmo_id key is there.
+            assert len(person) == 1
+
+    # Compare the old and new people lists without the new entry, they should match.
+    new = sorted(
+        [dict(p) for p in file_config.config["people"] if p["bmo_id"] != 3],
+        key=itemgetter("bmo_id"),
+    )
+    original = sorted([p for p in config["people"]], key=itemgetter("bmo_id"))
+    assert new == original
+
+    # Run clean without refresh.
+    clean(file_config, refresh=False)
+
+    # It is expected that only the new entry is updated, and the rest are intact.
+    cleaned = sorted(file_config.config["people"], key=itemgetter("bmo_id"))
+    assert len(cleaned) == 5
+    assert cleaned[0] == original[0]
+    assert cleaned[1] == original[1]
+    assert cleaned[2] == original[2]
+    assert cleaned[3] == {"bmo_id": 3, "name": "seven", "nick": "nanotubes"}
+
+
+@mock.patch("mots.config.get_bmo_data")
+def test_clean_added_user_with_refresh(get_bmo_data, repo, config, test_data):
+    """Test that updated BMO data is reflected when cleaning with a refresh."""
+    get_bmo_data.return_value = test_data
+
+    file_config = FileConfig(repo / "mots.yml")
+    file_config.load()
+
+    assert file_config.config["people"] == config["people"]
+
+    file_config.config["people"].append({"bmo_id": 3})
+    file_config.write()
+
+    assert len(file_config.config["people"]) == 5
+
+    # Check that the new entry is unaffected after write.
+    for person in file_config.config["people"]:
+        if person["bmo_id"] == 3:
+            # Ensure only the bmo_id key is there.
+            assert len(person) == 1
+
+    # Compare the old and new people lists without the new entry, they should match.
+    new = sorted(
+        [dict(p) for p in file_config.config["people"] if p["bmo_id"] != 3],
+        key=itemgetter("bmo_id"),
+    )
+    original = sorted([p for p in config["people"]], key=itemgetter("bmo_id"))
+    assert new == original
+
+    # Run clean with refresh.
+    clean(file_config, refresh=True)
+
+    # It is expected that all people will have been updated via BMO.
+    cleaned = sorted(file_config.config["people"], key=itemgetter("bmo_id"))
+    assert len(cleaned) == 5
+    assert cleaned[0] == {"bmo_id": 0, "name": "janeway", "nick": "captain"}
+    assert cleaned[1] == {"bmo_id": 1, "name": "tuvok", "nick": "2vk"}
+    assert cleaned[2] == {"bmo_id": 2, "name": "neelix", "nick": "cooks4u"}
+    assert cleaned[3] == {"bmo_id": 3, "name": "seven", "nick": "nanotubes"}
```

Comment:
Since `len(cleaned) == 5` I would expect we should also be checking `cleaned[4]` is a value we expect, right?

Reply:
Since the added (net-new) user is the last element of `cleaned`, we don't really need to check its values since there was no old value that would have changed (and the scope of this test is to ensure that running clean with newly added people + the refresh flag will indeed refresh the existing people entries.)

However, it wouldn't hurt to add this anyway, I can add a little comment to explain as well.